rails/ai-rules/rails-rules.md

```diff
@@ -0,0 +1,71 @@
+You are an expert in Ruby on Rails, PostgreSQL, and Hotwire (Turbo and Stimulus).
+
+Code Style and Structure
+- Follow thoughtbot's guide to write idiomatic Ruby https://github.com/thoughtbot/guides/tree/main/ruby
+- Follow Rails conventions and best practices.
+- Use object-oriented and functional programming patterns as appropriate.
+- Prefer iteration and modularization over code duplication.
+- Use descriptive variable and method names (e.g., user_signed_in?, calculate_total).
+- Structure files according to Rails conventions (MVC, concerns, helpers, etc.).
+
+Naming Conventions
+- Use snake_case for file names, method names, and variables.
+- Use CamelCase for class and module names.
+- Follow Rails naming conventions for models, controllers, and views.
+
+Ruby and Rails Usage
+- Use Ruby 3.x features when appropriate (e.g., pattern matching, endless methods).
+- Leverage Rails' built-in helpers and methods.
+- Use ActiveRecord effectively for database operations.
+
+Syntax and Formatting
+- Follow the Ruby Style Guide (https://rubystyle.guide/)
+- Use Ruby's expressive syntax (e.g., unless, ||=, &.)
+- Prefer double quotes for strings.
+
+Error Handling and Validation
+- Use exceptions for exceptional cases, not for control flow.
+- Implement proper error logging and user-friendly messages.
+- Use ActiveModel validations in models.
+- Handle errors gracefully in controllers and display appropriate flash messages.
+
+UI and Styling
+- Use Hotwire (Turbo and Stimulus) for dynamic, SPA-like interactions.
+- Use Rails view helpers and partials to keep views DRY.
+
+Performance Optimization
+- Use database indexing effectively.
+- Implement caching strategies (fragment caching, Russian Doll caching).
+- Use eager loading to avoid N+1 queries.
+- Optimize database queries using includes, joins, or select.
+
+Key Conventions
+- Follow RESTful routing conventions.
+- Use concerns for shared behavior across models or controllers.
+- Implement service objects for complex business logic.
```

Comment:
I'm not 100% sure about this for example. Lately I've been more leaning towards [Service objects are poorly-named models](https://dimiterpetrov.com/blog/service-objects-are-poorly-named-models/).